examples_tests/20.Megatexture/main.cpp

```diff
@@ -127,11 +127,15 @@ STextureData getTextureData(const asset::ICPUImage* _img, asset::ICPUVirtualText
 {
     const auto& extent = _img->getCreationParameters().extent;
 
+    auto imgAndOrigSz = asset::ICPUVirtualTexture::createPoTPaddedSquareImageWithMipLevels(_img, _uwrap, _vwrap);
+
     asset::IImage::SSubresourceRange subres;
     subres.baseMipLevel = 0u;
     subres.levelCount = core::findLSB(core::roundDownToPoT<uint32_t>(std::max(extent.width, extent.height))) + 1;
 
-    return _vt->pack(_img, subres, _uwrap, _vwrap, _borderColor);
+    auto addr = _vt->alloc(_img->getCreationParameters().format, imgAndOrigSz.second, subres, _uwrap, _vwrap);
+    _vt->commit(addr, imgAndOrigSz.first.get(), imgAndOrigSz.second, subres, _uwrap, _vwrap, _borderColor);
```

Comment:
addr already holds `imgAndOrigSz.second` as well as ` _uwrap, _vwrap` right?

Reply:
almost, STextureData holds `STextureData::E_WRAP_MODE`:
```
        enum E_WRAP_MODE
        {
            EWM_REPEAT = 0b00,
            EWM_CLAMP = 0b01,
            EWM_MIRROR = 0b10,
            EWM_INVALID = 0b11
        };
```
`ISampler::E_TEXTURE_CLAMP` maps to `E_WRAP_MODE` like this:
```
        static E_WRAP_MODE ETC_to_EWM(ISampler::E_TEXTURE_CLAMP _etc)
        {
            switch (_etc)
            {
            case ISampler::ETC_REPEAT:
                return STextureData::EWM_REPEAT;
            case ISampler::ETC_CLAMP_TO_EDGE:
            case ISampler::ETC_CLAMP_TO_BORDER:
                return STextureData::EWM_CLAMP;
            case ISampler::ETC_MIRROR:
            case ISampler::ETC_MIRROR_CLAMP_TO_EDGE:
            case ISampler::ETC_MIRROR_CLAMP_TO_BORDER:
                return STextureData::EWM_MIRROR;
            default:
                return STextureData::EWM_INVALID;
            }
        }
```
(clamp to edge and to border becomes one thing, and mirror and all mirror+clamp becomes one thing)
Ofc we can map back but it's lossy then. Mapping back is needed for padded copy. It would mean that padded copy will never use CLAMP_TO_BORDER (CLAMP_TO_EDGE instead) and will never use `MIRROR_CLAMP_*` as well (MIRROR instead)

Reply:
ah yes, you're right, in that case we need to assert that the `_uwrap, _vwrap` are compatible with the wrapping modes of the `SMasterTextureData`